custom_components/mqtt_discoverystream/__init__.py

```diff
@@ -112,8 +108,9 @@ async def message_received(msg):
         entity = explode_topic[2]
         element = explode_topic[3]
 
-        # Only handle service calls for discoveries we published
-        if f"{domain}.{entity}" not in hass.data[DOMAIN][discovery_topic]["conf_published"]:
+        # Only handle service calls for discoveries we published (or intend to publish)
+        if (f"{domain}.{entity}" not in hass.data[DOMAIN][discovery_topic]["conf_published"]
+            and not publish_filter(f"{domain}.{entity}")):
```

Comment:
Added this check to handle entities that don't change state until toggled (buttons, input_buttons, scripts, etc) because otherwise you have to manually run the entity on the primary Home Assistant server every restart before this integration will accept commands on those topics from the secondary instance(s).